internal/contour/listener.go

```diff
@@ -262,7 +269,17 @@ func tlscontext(secret *v1.Secret, alpnprotos ...string) *auth.DownstreamTlsCont
 	}
 }
 
-func httpfilter(routename string) listener.Filter {
+func httpfilter(routename string, useGrpcWeb bool) listener.Filter {
```

Comment:
Please don't add a parameter here. Add a new helper, say `grpcwebfilter` which sets the appropriate settings.